datasets/nchlt/nchlt.py

```diff
@@ -0,0 +1,174 @@
+        ),
+        NCHLTConfig(
+            name="tn",
+            version=datasets.Version("1.0.0"),
+            description="NCHLT Setswana Named Entity Annotated Corpus",
+            zip_file="https://repo.sadilar.org/bitstream/handle/20.500.12185/341/nchlt_setswana_named_entity_annotated_corpus.zip",
+        ),
+        NCHLTConfig(
+            name="ss",
+            version=datasets.Version("1.0.0"),
+            description="NCHLT Siswati Named Entity Annotated Corpus",
+            zip_file="https://repo.sadilar.org/bitstream/handle/20.500.12185/346/nchlt_siswati_named_entity_annotated_corpus.zip",
+        ),
+        NCHLTConfig(
+            name="ve",
+            version=datasets.Version("1.0.0"),
+            description="NCHLT Tshivenda Named Entity Annotated Corpus",
+            zip_file="https://repo.sadilar.org/bitstream/handle/20.500.12185/355/nchlt_tshivenda_named_entity_annotated_corpus.zip",
+        ),
+        NCHLTConfig(
+            name="ts",
+            version=datasets.Version("1.0.0"),
+            description="NCHLT Xitsonga Named Entity Annotated Corpus",
+            zip_file="https://repo.sadilar.org/bitstream/handle/20.500.12185/355/nchlt_tshivenda_named_entity_annotated_corpus.zip",
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "ner_tags": datasets.Sequence(
+                        datasets.features.ClassLabel(
+                            names=[
+                                "OUT",
+                                "B-PERS",
+                                "I-PERS",
+                                "B-ORG",
+                                "I-ORG",
+                                "B-LOC",
+                                "I-LOC",
+                                "B-MISC",
+                                "I-MISC",
+                            ]
+                        )
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://www.aclweb.org/anthology/W02-2024/",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        dl_dir = dl_manager.download_and_extract(self.config.zip_file)
+        all_filenames = set()
+        for root, dirs, files in os.walk(dl_dir):
+            for filename in files:
+                if filename.endswith(".txt") and "readme.txt" not in filename.lower():
+                    all_filenames.add(os.path.join(dl_dir, root, filename))
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filenames": all_filenames}),
+        ]
+
+    def _generate_examples(self, filenames):
+        for filename in filenames:
```

Comment:
```suggestion
        for filename in sorted(filenames):
```
we need to make the order of the iteration here deterministic since `os.walk` may differ depending of the environment

Reply:
Didn't know it was not deterministic :D